QueryNode cannot found tSafer

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: lastest
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

2021-12-01T15:46:51.621565532Z stdout F [2021/12/01 15:46:51.617 +00:00] [ERROR] [query_collection.go:538] ["[should not happen!] stop doUnsolvedMsg, err = cannot found tSafer, vChannel = by-dev-rootcoord-dml_103_429489809257136129v1"] [stack="github.com/milvus-io/milvus/internal/querynode.(*queryCollection).doUnsolvedQueryMsg\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/query_collection.go:538"]

### Expected Behavior

should not happen indeed should not happen!

### Steps To Reproduce

```markdown
Please check any jenkins ci log.
```


### Anything else?

_No response_

--- a/internal/querynode/task.go
+++ b/internal/querynode/task.go
@@ -787,6 +787,16 @@ func (r *releaseCollectionTask) Execute(ctx context.Context) error {
 	log.Debug("Execute release collection task", zap.Any("collectionID", r.req.CollectionID))
 	errMsg := "release collection failed, collectionID = " + strconv.FormatInt(r.req.CollectionID, 10) + ", err = "
 	log.Debug("release streaming", zap.Any("collectionID", r.req.CollectionID))
+	// sleep to wait for query tasks done
+	const gracefulReleaseTime = 1
+	time.Sleep(gracefulReleaseTime * time.Second)
+	log.Debug("Starting release collection...",
+		zap.Any("collectionID", r.req.CollectionID),
+	)
+
+	// remove query collection
+	r.node.queryService.stopQueryCollection(r.req.CollectionID)
+
 	err := r.releaseReplica(r.node.streaming.replica, replicaStreaming)
 	if err != nil {
 		return errors.New(errMsg + err.Error())
@@ -799,8 +809,6 @@ func (r *releaseCollectionTask) Execute(ctx context.Context) error {
 		return errors.New(errMsg + err.Error())
 	}
 	r.node.historical.removeGlobalSegmentIDsByCollectionID(r.req.CollectionID)
-	// remove query collection
-	r.node.queryService.stopQueryCollection(r.req.CollectionID)
 
 	log.Debug("ReleaseCollection done", zap.Int64("collectionID", r.req.CollectionID))
 	return nil
@@ -815,12 +823,6 @@ func (r *releaseCollectionTask) releaseReplica(replica ReplicaInterface, replica
 	log.Debug("set release time", zap.Any("collectionID", r.req.CollectionID))
 	collection.setReleaseTime(r.req.Base.Timestamp)
 
-	// sleep to wait for query tasks done
-	const gracefulReleaseTime = 1
-	time.Sleep(gracefulReleaseTime * time.Second)
-	log.Debug("Starting release collection...",
-		zap.Any("collectionID", r.req.CollectionID),
-	)
 	if replicaType == replicaStreaming {
 		r.node.dataSyncService.removeCollectionFlowGraph(r.req.CollectionID)
 		// remove partition flow graphs which partitions belong to the target collection

